tests/utils/stubs/service.stubs.ts

```diff
@@ -1,94 +1,82 @@
-    async encrypt ({ value }: HashGenerator.Params): Promise<HashGenerator.Result> {
-      return 'hashed_value'
+    async loadByEmail (email: string): Promise<LoadGuardianByEmailRepository.Result> {
+      return this.fakeGuardianData
     }
-  }
-  return new HashComparerStub()
-}
 
-const makeTokenGenerator = (): TokenGenerator => {
-  class TokenGeneratorStub implements TokenGenerator {
-    async generate (payload: any): Promise<string> {
-      return 'any_token'
+    async loadById (id: string): Promise<LoadGuardianByIdRepository.Result> {
+      return this.fakeGuardianData
     }
   }
-  return new TokenGeneratorStub()
+  return new GuardianRepositoryStub(fakeGuardianData)
 }
 
-const makeTokenDecoder = (): TokenDecoder => {
-  class TokenDecoderStub implements TokenDecoder {
-    async decode (token: TokenDecoder.Params): Promise<TokenDecoder.Result> {
-      return makeFakePayload()
+const makeHashService = (): HashGenerator & HashComparer => {
+  class HashServiceStub implements HashGenerator, HashComparer {
+    async compare (input: HashComparer.Params): Promise<boolean> {
+      return true
     }
-  }
-  return new TokenDecoderStub()
-}
 
-const makeLoadGuardianById = (): LoadGuardianByIdRepository => {
-  class LoadGuardianByIdStub implements LoadGuardianByIdRepository {
-    async loadById (id: LoadGuardianByIdRepository.Params): Promise<LoadGuardianByIdRepository.Result> {
-      return makeFakeGuardianData() as Guardian
+    async encrypt (input: HashGenerator.Params): Promise<string> {
+      return 'hashed_value'
     }
   }
-  return new LoadGuardianByIdStub()
+  return new HashServiceStub()
 }
 
-const makeLoadGuardianByEmail = (data: Guardian): LoadGuardianByEmailRepository => {
-  class LoadGuardianByEmailRepositoryStub implements LoadGuardianByEmailRepository {
-    async loadByEmail (email: LoadGuardianByEmailRepository.Params): Promise<LoadGuardianByEmailRepository.Result> {
-      return data
+const makeTokenService = (): TokenGenerator & TokenDecoder => {
+  class TokenServiceStub implements TokenGenerator, TokenDecoder {
+    async generate (payload: any): Promise<string> {
+      return 'any_token'
     }
-  }
-  return new LoadGuardianByEmailRepositoryStub()
-}
 
-const makeUpdateAccessTokenRepository = (): UpdateAccessTokenRepository => {
-  class UpdateAccessTokenRepositoryStub implements UpdateAccessTokenRepository {
-    async updateAccessToken (authentication: UpdateAccessTokenRepository.Params): Promise<UpdateAccessTokenRepository.Result> {
-      return true
+    async decode (token: TokenDecoder.Params): Promise<TokenDecoder.Result> {
+      return makeFakePayload()
     }
   }
-  return new UpdateAccessTokenRepositoryStub()
+  return new TokenServiceStub()
 }
 
 export {
-  makeEncrypter,
-  makeHashComparer,
-  makeHashGenerator,
-  makeTokenGenerator,
-  makeTokenDecoder,
-  makeLoadGuardianById,
-  makeLoadGuardianByEmail,
-  makeUpdateAccessTokenRepository
+  makeTokenService,
```

Comment:
I like this implementation and it looks like it cuts it down quite a bit, but I also made some changes to those files and we're likely to have conflict issues. You can refactor to `MakeFakeTokenService` so we can see that it's a fake service, so we differentiate on export because we use that pattern on factories too. What do you think?

Reply:
in this case it would be for all these services***

Reply:
Sure, make sense, that is a good name. I will change.